File: niuniu-take-out-springboot3/server/src/main/java/common/niuniu/service/AddressBookService.java
package common.niuniu.service;

import common.niuniu.entity.AddressBook;

import java.util.List;

public interface AddressBookService {
    void addAddress(AddressBook addressBook);

    List<AddressBook> list(AddressBook addressBook);

    void updateAddress(AddressBook addressBook);

    AddressBook getById(Integer id);

    void setDefault(AddressBook addressBook);

    void deleteById(Integer id);
}

File: niuniu-take-out-kotlin/server/src/main/kotlin/common/niuniu/service/AddressBookService.kt
package common.niuniu.service

import common.niuniu.po.AddressBook

interface AddressBookService {
    fun addAddress(addressBook: AddressBook?)

    fun list(addressBook: AddressBook): List<AddressBook>

    fun updateAddress(addressBook: AddressBook?)

    fun getById(id: Int?): AddressBook?

    fun setDefault(addressBook: AddressBook?)

    fun deleteById(id: Int?)
}
